### PR TITLE
Create a User model

### PR DESCRIPTION
**Pull request by MichelleOha**
I have created the users model with the 2 required fields. Run rails DB migrate. 

https://trello.com/c/0HLa8uT6/7-create-user-for-payment

1st push - This is the first task for this Trello ticket.
2nd push - Associate the user to payment (user_id is null: true to be optional as I have already created payments in the table prior to creating users)
3rd push - Add model method to return full name 
4th push - resolve conflicts

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -4,5 +4,5 @@
 
   # Defines the root path route ("/")
   # root "posts#index"
-  resources :payments, only: [ :index, :create, :show], param: :id
+  resources :payments, only: [ :index, :create, :show]
```
Comment (JosephA91) [inline config/routes.rb]: Prefer to format like:
`resources :payments, only: [:index, :create, :show]`

File: db/migrate/20240308005744_make_user_id_nullable_in_payments.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# db/migrate/20240308005744_make_user_id_nullable_in_payments.rb
+
+class MakeUserIdNullableInPayments < ActiveRecord::Migration[7.1]
+  def change
+    change_column :payments, :user_id, :integer, null: true
```
Comment (JosephA91) [inline db/migrate/20240308005744_make_user_id_nullable_in_payments.rb]: My preference is to create the reference like this including and index:
`add_reference :payments, :user, foreign_key: true, null: true, index: true`

To change this you need to run locally:
```
rails db:rollback
# change the contents of the migration from change_column to add_reference as listed about, then run the migration below
rails db:migrate
```

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_many :payments
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+
+  # Method to return full_name
+  def full_name
+    "#{first_name} #{last_name}"
+  end
```
Comment (JosephA91) [inline app/models/user.rb:10]: Nice one!

**Review by JosephA91 — COMMENTED** [review]
Some comments for you, @MichelleOha - the solution is looking good.

File: config/routes.rb (hunk 1 above)
Comment (MichelleOha) [reply config/routes.rb]: amended, thanks Joe

File: db/migrate/20240308005744_make_user_id_nullable_in_payments.rb (hunk 2 above)
Comment (MichelleOha) [reply db/migrate/20240308005744_make_user_id_nullable_in_payments.rb]: Ok, so I came across a few hurdles trying to get this to work. It seems to be working now (as I haven't received an error message 🤣). Would you mind checking this for me to ensure I have done it correctly please? @anandjacobmable @JosephA91 Thanks 🫶🏼

**Review by JosephA91 — APPROVED** [review]

**Review by anandjacobmable — APPROVED** [review]
nice work  👍

**Pull request merged**